File: src/main/java/com/kishan/java/leetcode/juneChallenges/TreeNode.java
package com.kishan.java.leetcode.juneChallenges;

import sun.reflect.generics.tree.Tree;

public class TreeNode {
    private int value;
    private TreeNode left;
    private TreeNode right;

    public TreeNode() {

    }

    public TreeNode(int value) {
        this();
        this.value = value;
    }

    public TreeNode(int value, TreeNode left) {
        this(value);
        this.left = left;
    }

    public TreeNode(int value, TreeNode left, TreeNode right) {
        this(value, left);
        this.right = right;
    }

    public int getValue() {
        return value;
    }

    public void setValue(int value) {
        this.value = value;
    }

    public void setLeft(TreeNode left) {
        this.left = left;
    }

    public void setRight(TreeNode right) {
        this.right = right;
    }

    public TreeNode getLeft() {
        return left;
    }

    public TreeNode getRight() {
        return right;
    }
}

File: src/main/scala/com/kishan/scala/leetcode/juneChallenges/TreeNode.scala
package com.kishan.scala.leetcode.juneChallenges

/*
* toString Method of the Class have been override order to provide string represenation of the BST
*
* */

//Definition for a binary tree node.
class TreeNode(_value: Int = 0, _left: TreeNode = null, _right: TreeNode = null) {
  var value: Int = _value
  var left: TreeNode = _left
  var right: TreeNode = _right

  override def toString: String = {
    var result = s"${this.value}"
    if (this.left != null) {
      result = s"${this.left.toString} <- " + result
    }
    if (this.right != null) {
      result = result + s" -> ${this.right.toString}"
    }
    "[" + result + "]"
  }
}
